src/Data/Nat/Induction.agda

```diff
@@ -85,8 +86,16 @@ module _ {ℓ} where
 <-Rec : ∀ {ℓ} → RecStruct ℕ ℓ ℓ
 <-Rec = WfRec _<_
 
-<-wellFounded : WellFounded _<_
-<-wellFounded = Subrelation.wellFounded ≤⇒≤′ <′-wellFounded
+mutual
+
+  <-wellFounded : WellFounded _<_
+  <-wellFounded n = acc (<-wellFounded′ n)
+
+  <-wellFounded′ : ∀ n → <-Rec (Acc _<_) n
+  <-wellFounded′ zero    y ()
+  <-wellFounded′ (suc n) y y<1+n with <-wellFounded n | m<1+n⇒m<n∨m≡n y<1+n
+  ... | wfn@(acc rec) | inj₁ y<n  = rec y y<n
```

Comment:
No need for the `@` here.

Reply:
The original used `acc-inverse`; so I've inlined that pattern analysis; I retained the (unused) pattern variable `wfn` to show the sharing of the recursive call between the two case; but the first case needs to further decompose that proof, in order to appeal to accessibility. Or did you mean something else?

Reply:
Yes I simply meant that it was unused. Personally I think it's obvious that the two expressions in the two cases refer to the same pattern match simply by the `|` syntax, but if you feel strongly that it should be kept that's fine.